extension/src/paymentHandler/make-payment.handler.js

```diff
@@ -0,0 +1,40 @@
+const { makePayment } = require('../web-component-service')
+const pU = require('./payment-utils')
+const c = require('../config/constants')
+
+async function execute (paymentObject) {
+  const makePaymentRequestObj = JSON.parse(paymentObject.custom.fields.makePaymentRequest)
+  // todo: check amount planned
+  const { request, response } = await makePayment(makePaymentRequestObj)
+  const actions = [
+    pU.createAddInterfaceInteractionAction({
+      request, response, type: c.CTP_INTERACTION_TYPE_MAKE_PAYMENT
+    }),
+    pU.createSetCustomFieldAction(c.CTP_CUSTOM_FIELD_MAKE_PAYMENT_RESPONSE, response)
+  ]
+
+  // eslint-disable-next-line default-case
+  switch (response.resultCode) {
+    case 'Authorised':
+      actions.push(
+        pU.createAddTransactionAction('Authorization', 'Success',
+          paymentObject.amountPlanned.centAmount, paymentObject.amountPlanned.currencyCode,
+          response.pspReference)
+      )
+      break
+    case 'Refused':
+    case 'Error':
+      actions.push(
```

Comment:
Should we consider the other result codes like Cancelled ?

Reply:
I also had this question. So far I could not simulate the response code from Cancelled, probably I need to ping Adyen again to provide me with the result. 

But if the user cancels the payment, then there's no Transaction. Also we didn't describe Cancelled code in our task, so I guess we also thought like this. And we don't have Cancelled code, only `CancelAuthorization`, which has slightly different meaning: https://docs.commercetools.com/http-api-projects-payments#transactionstate